### ISSUE TITLE
Getting lots of 'context deadline exceeded' with new OpenAI models

### ISSUE THREAD

**Issue by adambeer** [weaviate/weaviate#4131]
### How to reproduce this bug?

Using the Javascript/TS library (v2.0.0) and doing batch ingestion.
Using the model 'text-embedding-3-large' with 1024 dimensions. We also tried 3072 dimensions as well.
The issue seems to happen intermittently.

### What is the expected behavior?

No errors.

### What is the actual behavior?

Were noticing a lot of the following errors:

''update vector: send POST request: Post "https://api.openai.com/v1/embeddings": context deadline exceeded (Client.Timeout exceeded while awaiting headers)'

We never got this type of error with ada-002 and it seems to happen randomly on various objects in different batches.

### Server Version

1.23.7

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by etiennedi** [comment]
Thanks. `context deadline exceeded` is just the Go way of saying "timeout". Sounds like OpenAI's APIs are timing out. Still thanks for flagging though, we can investigate if we are doing something wrong (e.g. sending too much data at the same time) or if this is entirely on their end.

I believe the timeout was also configureable? Correct, @antas-marcin? There was an env var IIRC?

**Comment by antas-marcin** [comment]
Yes this there is `MODULES_CLIENT_TIMEOUT` environment setting @adambeer if you want to increase this value to 2 minutes just set `MODULES_CLIENT_TIMEOUT=2m`

**Comment by antas-marcin** [comment]
@adambeer the default timeout set for `MODULES_CLIENT_TIMEOUT` is `50 seconds`.
We are now working also on mitigating this problem by adding batch support for module system. I will let you know about our progress, this should feature land in `v1.24`.

**Comment by antas-marcin** [comment]
@adambeer during our work on batch support for module system we have also experienced timeouts even if we have been sending only a few objects in request this is also something that we didn't observe before. This leads me to a conclusion that maybe there's actually something on OpenAI's side, a problem that only occurs when using their new V3 models?

**Comment by adambeer** [comment]
In total I have been getting 4 different errors while using the new openAI model:

1. context deadline exceeded
2. fetch failed (latest JS/TS lib)
3. 503 Service Unavailable
4. 500

It seems they are having some big issues over there right now.

**Comment by adambeer** [comment]
@antas-marcin @etiennedi I tried the preview image of this with _MODULES_CLIENT_TIMEOUT: '2m'_ in my docker-compose.yml environment and am getting an excessive amount of "context deadline exceeded" errors. Am I missing something here?

**Comment by dudanogueira** [comment]
This should be fixed now. If not, please, open a new issue. thanks!
